core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -200,12 +206,56 @@ public BuddyDto addBuddyToAcceptingUser(UserDto acceptingUser, BuddyConnectReque
 				() -> UserServiceException.notFoundById(connectRequestMessageEntity.getSenderUserId()));
 
 		acceptingUser.assertMobileNumberConfirmed();
-		Buddy buddy = createBuddyEntity(connectRequestMessageEntity);
-		BuddyDto buddyDto = BuddyDto.createInstance(buddyRepository.save(buddy));
+		Buddy buddy = buddyRepository.save(createBuddyEntity(connectRequestMessageEntity));
+		BuddyDto buddyDto = createBuddyDtoForNewBuddy(buddy);
 		userService.addBuddy(acceptingUser, buddyDto);
 		return buddyDto;
 	}
 
+	private BuddyDto createBuddyDto(Buddy buddy)
+	{
+		Optional<UserAnonymizedDto> buddyUserAnonymizedDto = getUserAnonymizedDto(buddy);
+		BuddyAnonymizedDto buddyAnonymizedDto = getBuddyAnonymized(buddy);
+		return BuddyDto.createInstance(buddy, buddyAnonymizedDto, buddyUserAnonymizedDto);
+	}
+
+	private BuddyDto createBuddyDtoForNewBuddy(Buddy buddy)
+	{
+		Optional<UserAnonymizedDto> buddyUserAnonymizedDto = buddy.getUserAnonymizedId()
+				.map(userAnonymizedService::getUserAnonymized);
+		BuddyAnonymizedDto buddyAnonymizedDto = BuddyAnonymizedDto.createInstance(buddy.getBuddyAnonymized());
+		return BuddyDto.createInstance(buddy, buddyAnonymizedDto, buddyUserAnonymizedDto);
+	}
+
+	private BuddyAnonymizedDto getBuddyAnonymized(Buddy buddy)
+	{
+		return getBuddyAnonymized(buddy, getOwningUserAnonymized(buddy));
+	}
+
+	private BuddyAnonymizedDto getBuddyAnonymized(Buddy buddy, UserAnonymizedDto owningUserAnonymizedDto)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "owningUserAnonymizedDto". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1172)